main.sh

```diff
@@ -91,12 +92,27 @@ trap cleanup_before_exit EXIT
 
 # Translate usage string -> getopts arguments, and set $arg_<flag> defaults
 while read line; do
-  opt="$(echo "${line}" |awk '{print $1}' |sed -e 's#^-##')"
+  # fetch single character version of option sting
+  opt="$(echo "${line}" |awk '{print $1}' |sed -e 's#^-\(.\)#\1#')"
```

Comment:
Please note: This is the only place where I have changed or added the use of `sed`. This change is portable across OSes, but, I like your suggestion of using only awk which tends to be more forgiving across different OSes.